freqtrade/optimize/hyperopt.py

```diff
@@ -410,6 +410,35 @@ def prepare_hyperopt_data(self) -> None:
         # Store non-trimmed data - will be trimmed after signal generation.
         dump(preprocessed, self.data_pickle_file)
 
+    def get_asked_points(self, n_points: int) -> List[List[Any]]:
+        '''
+        Enforce points returned from `self.opt.ask` have not been already evaluated
+
+        Steps:
+        1. Try to get points using `self.opt.ask` first
+        2. Discard the points that have already been evaluated
+        3. Retry using `self.opt.ask` up to 3 times
+        4. If still some points are missing in respect to `n_points`, random sample some points
+        5. Repeat until at least `n_points` points in the `asked_non_tried` list
+        6. Return a list with length truncated at `n_points`
+        '''
+        i = 0
+        asked_non_tried: List[List[Any]] = []
+        while i < 100 and len(asked_non_tried) < n_points:
+            if i < 3:
+                self.opt.cache_ = {}
```

Comment:
Did you also observe if this also improved resuts - or if they got worse again? 

my suspicion here is that it may be beneficial to use completely random points once you hit this ... which is what happened before this line was added - where you described that results were better ...

Reply:
I'm not seeing a big difference, to be honest. My main issue was that, after a certain point, the ask method keeps returning the same point (or few points) over and over again because it doesn't have any interest in exploring other strategies to get to the minimum. I found it to be an issue, especially when using just a few parameters.

The downside of this approach is that right now it's not clear when the optimizer is running random points and when it's running 'optimal' points. It'd be great to store this information and display it in the progress table somehow.

Reply:
is there a benefit to knowing this? (it's something you can't influence anyway) ... 

i mean - if my results are "best" based on random points or "real optimization" ... do i really care?